google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -94,11 +119,78 @@ def log_metrics(self, metrics: Dict[str, Union[str, float, int]]):
+
+            Example:
+
+            aiplatform.init(experiment='exp-1')
+            aiplatform.set_run(run='run-1')
+            aiplatform.log_params({'learning_rate': 0.1})
+            aiplatform.log_metrics({'accuracy': 0.9})
+
+            aiplatform.set_run(run='run-2')
+            aiplatform.log_params({'learning_rate': 0.2})
+            aiplatform.log_metrics({'accuracy': 0.95})
+
+            Will result in the following DataFrame
+            ___________________________________________________________________________
+            | experiment_name | run_name      | param.learning_rate | metric.accuracy |
+            ---------------------------------------------------------------------------
+            | exp-1           | run-1         | 0.1                 | 0.9             |
+            | exp-1           | run-2         | 0.2                 | 0.95            |
+            ---------------------------------------------------------------------------
+
+            Args:
+              experiment: Name of the Experiment to filter results.
+
+            Returns:
+              Pandas Dataframe of Experiment with metrics and parameters.
+            """
+
+        # TODO remove this extra API call once our BE can support project ID in filters.
+        try:
+            experiment_context = _Context(resource_name=experiment)
+        except exceptions.NotFound:
+            logging.error(f"Experiment {experiment} not found.")
+            return
+        if experiment_context.schema_title != constants.SYSTEM_EXPERIMENT:
+            raise ValueError(
+                f"Please provide a valid experiment name. {experiment} is not a experiment."
+            )
+
+        return self._query_runs_to_data_frame(
+            context_name=experiment,
+            context_resource_name=experiment_context.resource_name,
+            source="experiment",
+        )
 
-    def get_pipeline(self, pipeline: str):
-        raise NotImplementedError("get_pipeline not implemented")
+    def get_pipeline(self, pipeline: str) -> pd.DataFrame:
+        """Returns a Pandas DataFrame of the parameters and metrics associated with one pipeline.
+
+                    Args:
+                      pipeline: Name of the Pipeline to filter results.
+
+                    Returns:
+                      Pandas Dataframe of Pipeline with metrics and parameters.
+                    """
+
+        # TODO remove this extra API call once our BE can support project ID in filters.
+        try:
+            pipeline_context = _Context(resource_name=pipeline)
+        except exceptions.NotFound:
+            logging.error(f"Pipeline {pipeline} not found.")
+            return
+        if pipeline_context.schema_title != constants.SYSTEM_PIPELINE:
+            raise ValueError(
+                f"Please provide a valid pipeline name. {pipeline} is not a pipeline."
+            )
```

Comment:
nit: can probably dedupe all this if you create a private _get_context or something similar and pass the resource_name, "pipeline"|"experiment" and the schema_title?

Reply:
Good point. refracted this part of code to a method.